batchspawner/batchspawner.py

```diff
@@ -429,15 +429,11 @@ class BatchSpawnerRegexStates(BatchSpawnerBase):
 
     def state_ispending(self):
         assert self.state_pending_re, "Misconfigured: define state_running_re"
-        if self.job_status and re.search(self.state_pending_re, self.job_status):
-            return True
-        else: return False
+        return bool(self.job_status and re.search(self.state_pending_re, self.job_status))
```

Comment:
I recommend leaving as currently written but clean up the `else` statement to put the return on a separate line. 

By adding the bool(), Python will not short-circuit the expression if self.job_status evaluates as false and will do the regex search even if not needed.

Reply:
Are you sure @willingc about the differently behaving "short-circuit" between `if` and `bool`? I am not aware of any difference in this case since in both cases it needs first to evaluate the conditional for its bool value, and thus it would short-circuit (stop evaluating as soon as the first one evaluates to False) in either of the cases... Since I trust no-one, including myself, here is a quick demo to support that:
```shell
$> cat /tmp/shortc.py 
def func(x):
    print("in func")
    return bool(x)

import sys

b = bool(int(sys.argv[1]))
v = int(sys.argv[2])
print("b=%r v=%r" % (b,v))

if b and func(v):
    print("if True")
else:
    print("if False")

print("bool %s" % bool(b and func(v)))

$> python3.7 /tmp/shortc.py 0 1
b=False v=1
if False
bool False

$> python3.7 /tmp/shortc.py 1 1
b=True v=1
in func
if True
in func
bool True

$> python3.7 /tmp/shortc.py 1 0
b=True v=0
in func
if False
in func
bool False

$> python3.7 /tmp/shortc.py 0 0
b=False v=0
if False
bool False
```
The only advantage I see if of the original way (and I will change back now, since I do not have strong opinion on this) is that you could easier monitor the coverage -- which of the cases was triggered by the tests code, and which possibly not, which isn't possible if it is just a one line expression